doc/release-notes-1.9.rst

```diff
@@ -10,8 +10,8 @@ We are pleased to announce the release of Zephyr kernel version 1.9.0
 
 Major enhancements planned with this release include:
 
-* POSIX API Layer
-* BSD Socket Support
+* Pthreads compatible API
+* BSD Sockets compatible API
```

Comment:
I'd say it should be either "Subset of BSD Sockets API" or "BSD Sockets compatible API". The latter is how it's called in the docs, so I used that.

I didn't follow Pthread patch closely, but Pthread is also vast API, and I doubt all of it was implemented right away (and I remember @andyross' RFC on how Zephyr threading might be change to accommodate Pthread model better). All in all, I used similar "compatible" formulation for consistency, though @andyross is in reviewers to see if that makes sense.

Overall, this is an RFC of what can be added to release notes, I'd say even further editorial effort is required on the whole notes anyway (e.g., I'd try to sort/group related things together - low-level stuff vs application protocols for example).